codebase/superdarn/src.bin/tk/tool/make_fit.1.19/make_fit.c

```diff
@@ -328,32 +482,31 @@ int main(int argc,char *argv[]) {
           /*FitacfFree(fit_prms);*/
         }
         else {
-          fprintf(stderr, "Unable to allocate fit_prms!\n");
-          exit(-1);
+            fprintf(stderr, "Unable to allocate fit_prms!\n");
+            free_radarstructs(network, prm, raw);
+            free_files(rawfp, fp, fitfp, inxfp);
+            free_fitstructs(fit_prms, fit, fblk);
+            exit(-1);
         }
       }
       else if (fitacf_version == 25) {
         FitACF(prm,raw,fblk,fit);
       }
       else {
-        fprintf(stderr, "The requested fitacf version does not exist\n");
-        exit(-1);
+            fprintf(stderr, "The requested fitacf version does not exist\n");
+            free_radarstructs(network, prm, raw);
+            free_files(rawfp, fp, fitfp, inxfp);
+            free_fitstructs(fit_prms, fit, fblk);
+            exit(-1);
       }
     }
 
 
   } while (status==0);
-  FitFree(fit);
-
-  if (fitacf_version == 30) {
-    FitacfFree(fit_prms);
-  }
-  else if (fitacf_version == 25) {
-    FitACFFree(fblk);
-  }
-  else {
 
-  }
+  free_radarstructs(network, prm, raw);
+  free_files(rawfp, fp, fitfp, inxfp);
+  free_fitstructs(fit_prms, fit, fblk);
 
   if (old) OldRawClose(rawfp);
```

Comment:
This line should be removed as `rawfp` would have already been freed by the call to `free_files` at line 508.